File: CS/ViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Xpf.Printing;
using DevExpress.XtraReports.UI;
using System;

namespace ValidateEditingFields_MVVM {
    public class ViewModel : ViewModelBase {
        public XtraReport Report { get; }
        public ViewModel() {
            Report = new XtraReport1();
        }
        
        [Command]
        public void OnShowingEditingFieldEditor(ShowingEditingFieldEditorEventArgs args) {
            if(args.EditingField.ID == "CheckField" || args.EditingField.ID == "ImageField")
                args.Cancel = true;
        }
        
        [Command]
        public void OnValidateEditingField(EditingFieldValidationEventArgs args) {
            if(args.EditingField.ID == "DateField")
                ValidateDateTimeEditingField(args);
        }

        void ValidateDateTimeEditingField(EditingFieldValidationEventArgs args) {
            DateTime value = default(DateTime);
            try {
                value = Convert.ToDateTime(args.Value);
            } catch (Exception e) {
                args.IsValid = false;
                args.ErrorMessage = e.Message;
                args.ErrorType = DevExpress.XtraEditors.DXErrorProvider.ErrorType.Critical;
            }
            
            DateTime now = DateTime.Now;
            args.IsValid = value.Year == now.Year && value.Month == now.Month;
        }
    }
}

File: VB/ViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Xpf.Printing
Imports DevExpress.XtraReports.UI
Imports System

Namespace ValidateEditingFields_MVVM
	Public Class ViewModel
		Inherits ViewModelBase

		Public ReadOnly Property Report() As XtraReport
		Public Sub New()
			Report = New XtraReport1()
		End Sub

		<Command>
		Public Sub OnShowingEditingFieldEditor(ByVal args As ShowingEditingFieldEditorEventArgs)
			If args.EditingField.ID = "CheckField" OrElse args.EditingField.ID = "ImageField" Then
				args.Cancel = True
			End If
		End Sub

		<Command>
		Public Sub OnValidateEditingField(ByVal args As EditingFieldValidationEventArgs)
			If args.EditingField.ID = "DateField" Then
				ValidateDateTimeEditingField(args)
			End If
		End Sub

		Private Sub ValidateDateTimeEditingField(ByVal args As EditingFieldValidationEventArgs)
			Dim value As Date = Date.MinValue
			Try
				value = Convert.ToDateTime(args.Value)
			Catch e As Exception
				args.IsValid = False
				args.ErrorMessage = e.Message
				args.ErrorType = DevExpress.XtraEditors.DXErrorProvider.ErrorType.Critical
			End Try

			Dim now As Date = Date.Now
			args.IsValid = value.Year = now.Year AndAlso value.Month = now.Month
		End Sub
	End Class
End Namespace
